server/src/handle_chaos_message.rs

```
@@ -0,0 +1,482 @@
+    fn from(proto: &chaos_proto::chaos::Severity) -> Severity {
+        Severity {
+            id: proto.get_id().to_string(),
+            wording: proto.get_wording().to_string(),
+            color: proto.get_color().to_string(),
+            priority: proto.get_priority() as u32,
+            effect: proto.get_effect().into(),
+        }
+    }
+}
+
+impl From<chaos_proto::gtfs_realtime::Alert_Effect> for Effect {
+    fn from(proto: chaos_proto::gtfs_realtime::Alert_Effect) -> Effect {
+        use chaos_proto::gtfs_realtime::Alert_Effect;
+        match proto {
+            Alert_Effect::NO_SERVICE => Effect::NoService,
+            Alert_Effect::REDUCED_SERVICE => Effect::ReducedService,
+            Alert_Effect::SIGNIFICANT_DELAYS => Effect::SignificantDelays,
+            Alert_Effect::DETOUR => Effect::Detour,
+            Alert_Effect::ADDITIONAL_SERVICE => Effect::AdditionalService,
+            Alert_Effect::MODIFIED_SERVICE => Effect::ModifiedService,
+            Alert_Effect::OTHER_EFFECT => Effect::OtherEffect,
+            Alert_Effect::UNKNOWN_EFFECT => Effect::UnknownEffect,
+            Alert_Effect::STOP_MOVED => Effect::StopMoved,
+        }
+    }
+}
+
+impl From<&chaos_proto::chaos::Cause> for Cause {
+    fn from(proto: &chaos_proto::chaos::Cause) -> Cause {
+        Cause {
+            id: proto.get_id().to_string(),
+            wording: proto.get_wording().to_string(),
+            category: proto.get_category().get_name().to_string(),
+            created_at: ts_to_dt(proto.get_created_at()),
+            updated_at: ts_to_dt(proto.get_created_at()),
+        }
+    }
+}
+
+impl From<&chaos_proto::gtfs_realtime::TimeRange> for DateTimePeriod {
+    fn from(proto: &chaos_proto::gtfs_realtime::TimeRange) -> DateTimePeriod {
+        DateTimePeriod {
+            start: NaiveDateTime::from_timestamp(proto.get_start() as i64, 0),
+            end: NaiveDateTime::from_timestamp(proto.get_end() as i64, 0),
```

proto.get_start() and proto.get_end() are u64 right ?
So the conversion to i64 may fail, and you should return an error when this happen